catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/settings/SettingsResource.java

```diff
@@ -0,0 +1,240 @@
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.json.JsonPatch;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.shared.utils.io.IOUtil;
+import org.openmetadata.catalog.CatalogApplicationConfig;
+import org.openmetadata.catalog.filter.EventFilter;
+import org.openmetadata.catalog.filter.FilterRegistry;
+import org.openmetadata.catalog.filter.Filters;
+import org.openmetadata.catalog.jdbi3.CollectionDAO;
+import org.openmetadata.catalog.jdbi3.SettingsRepository;
+import org.openmetadata.catalog.resources.Collection;
+import org.openmetadata.catalog.security.Authorizer;
+import org.openmetadata.catalog.settings.Settings;
+import org.openmetadata.catalog.settings.SettingsType;
+import org.openmetadata.catalog.util.EntityUtil;
+import org.openmetadata.catalog.util.JsonUtils;
+import org.openmetadata.catalog.util.ResultList;
+
+@Path("/v1/settings")
+@Api(value = "Settings Collection", tags = "Settings collection")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Collection(name = "settings")
+@Slf4j
+public class SettingsResource {
+  private final ScheduledExecutorService exec = Executors.newSingleThreadScheduledExecutor();
+  private final SettingsRepository settingsRepository;
+  private final Authorizer authorizer;
+
+  @SuppressWarnings("unused") // Method used for reflection
+  public void initialize(CatalogApplicationConfig config) throws IOException {
+    initSettings();
+  }
+
+  private void initSettings() throws IOException {
+    List<String> jsonDataFiles = EntityUtil.getJsonDataResources(".*json/data/settings/settingsData.json$");
+    if (jsonDataFiles.size() != 1) {
+      LOG.warn("Invalid number of jsonDataFiles {}. Only one expected.", jsonDataFiles.size());
+      return;
+    }
+    String jsonDataFile = jsonDataFiles.get(0);
+    try {
+      String json = IOUtil.toString(getClass().getClassLoader().getResourceAsStream(jsonDataFile));
+      List<Settings> settings = JsonUtils.readObjects(json, Settings.class);
+      settings.forEach(
+          (setting) -> {
+            try {
+              Settings storedSettings = settingsRepository.getConfigWithKey(setting.getConfigType().toString());
```

Comment:
what happens when there is change after the server start here.

Reply:
we were using a threaded call to db , which has been changed to use Guava cache like we use for Policy